prometheus/internalmetrics/internal_metrics.go

```diff
@@ -0,0 +1,202 @@
+// Package internalmetrics provides functionality to collect Prometheus metrics.
+package internalmetrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	// Because this package is used all throughout the codebase, be VERY careful adding new
+	// kiali imports here. Most likely you will encounter an import cycle error that will
+	// cause a compilation failure.
+)
+
+// These constants define the different label names for the different metric timeseries
+const (
+	labelGraphKind        = "graph_kind"
+	labelGraphType        = "graph_type"
+	labelWithServiceNodes = "with_service_nodes"
+	labelAppender         = "appender"
+	labelName             = "name"
+	labelQueryID          = "query_id"
+)
+
+// MetricsType defines all of Kiali's own internal metrics.
+type MetricsType struct {
+	GraphsGenerated          *prometheus.CounterVec
```

Comment:
I wonder if this counter is really useful, iirc when we define an histogram (for GraphGenerationTime) it also stores the count, so maybe we could use it as the counter?